core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -114,17 +114,17 @@ public String getName()
         if (!element.isNullable() && !metadata.getConnectorCapabilities(session, catalogName).contains(NOT_NULL_COLUMN_CONSTRAINT)) {
             throw semanticException(NOT_SUPPORTED, element, "Catalog '%s' does not support NOT NULL for column '%s'", catalogName.getCatalogName(), element.getName());
         }
-
-        Map<String, Expression> sqlProperties = mapFromProperties(element.getProperties());
+        if (hasDefaultProperty(element.getProperties())) {
+            throw semanticException(NOT_SUPPORTED, element, "Setting a column property to DEFAULT in an ALTER TABLE...ADD COLUMN... statement is not supported");
```

Comment:
It would be better to point to actual `DEFAULT` element, so that it's marked in CLI. Maybe add method like `throwOnDefaultProperty(..., message)`